### PR TITLE
Finish sql1 exercise

### PR DESCRIPTION
**Pull request by shaharshw**

### CODE REVIEW

File: src/main/resources/db/migration/V202408271754__test.sql [hunk 1]
```diff
@@ -1,5 +0,0 @@
-create table test_me
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408271754__test.sql]: do not delete existing migration, will cause problems in prod

File: src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+create table pets
+(
+    id              UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name            VARCHAR(255),
+    type            VARCHAR(255),
+    data_of_arrival DATE,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql]: you can add here default value

File: src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    company_id      UUID
+);
+
+create index idx_pets_company_id
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql]: great, but it's better to write it as one liner

File: src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql [hunk 4]
```diff
@@ -0,0 +1,25 @@
+create table owner
+(
+    id          UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name        VARCHAR(255),
+    company_id  UUID,
+    employee_id UUID
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql]: employee_id in bob is string

File: src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+);
+
+create index idx_owner_company_id_employee_id
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql]: same here

File: src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+
+/*
+    insert into pets(name, type, data_of_arrival, company_id)
+    values('Leo', 'Dog', '2022-09-08', 'f7b3b3b3-7b3b-4b3b-8b3b-3b3b3b3b3b3b');
+
+    insert into pets(name, type, data_of_arrival, company_id)
+    values('Tom', 'Cat', '2024-09-08', 'f7b3b3b3-7b3b-4b3b-8b3b-3b3b3b3b3b3b');
+
+    select * from pets where type = 'Dog';
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql]: you can use distinct

File: src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+    delete from pets where id = '8b3fe77a-d799-4684-ad3a-7794e4e25bb5';
+
+    select * from pets where data_of_arrival < '2023-09-08';
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql]: this is not the query - Get the pets that arrived more than a year ago
this query brings all pets that arrived before '2023-09-08'

**Review by mayhuberman — CHANGES_REQUESTED** [review]
👑

File: src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql (hunk 7 above)
Comment (shaharshw) [reply src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql]: Got it, I fixed this.

File: src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql [hunk 8]
```diff
@@ -0,0 +1,11 @@
+create table pets
+(
+    id              UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name            VARCHAR(255),
+    type            VARCHAR(255),
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql]: should be not null

File: src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    data_of_arrival DATE DEFAULT CURRENT_DATE,
+    company_id      UUID
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_pet_table_and_index.sql]: company_id in bob is bigint

File: src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql [hunk 10]
```diff
@@ -0,0 +1,22 @@
+create table owner
+(
+    id          UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name        VARCHAR(255),
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql]: should be not null

File: src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    company_id  UUID,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081500__create_owner_table_and_index.sql]: company_id in bob is bigint

**Pull request merged**